pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -189,6 +196,235 @@ func getV1Registries(config *tomlConfig) ([]Registry, error) {
 	return registries, nil
 }
 
+// registryIsMatch returns true if flags in the registry entry indicate that
+// its URL might not be used as-is, but will have URLs matched against it in
+// order to selectively apply configuration flags to an unlisted URL.
+func registryIsMatch(reg Registry) bool {
+	return reg.Insecure || reg.Blocked
+}
+
+// registryMatches returns true if the candidate location matches the pattern.
+// Recognized patterns include:
+// * IPv4[/class][:port][/path]
```

Comment:
Hum; `192.168.31.0/24` is a valid repository named `24` on the host `192.168.31.0`, so this is ambiguous.
(And `192.168.31.0/24:443` is an image tagged `443` in that repository!)

Also, the `Registry` object has not been designed with any patterns/groups in mind.

Arguably, the CIDR match is an useful feature that should be possible regardless — but then again, 1) this code does CIDR matches only against explicit IP addresses (rightly, because matching DNS names, at least with the Go TLS stack, can only work with a TOCTOU vulnerability), while Docker matches even DNS names, so it is not compatible anyway, and 2) marking whole networks insecure is just lazy and I we should not encourage that.

If this should be supported at all, it probably needs to be a separate `CIDRRange` type or something like that, not a `Registry` value.

@vrothberg WDYT?

Reply:
> If this should be supported at all, it probably needs to be a separate CIDRRange type or something like that, not a Registry value.

I agree. Docker has dedicated config switches for CIDRs (see https://github.com/vrothberg/moby/commit/b573b5e4d229887d486f94a39935f72d501588a5#diff-73caf4b121449e3cc9df8ab7554b31baL11).